lib/elixir/lib/enum.ex

```diff
@@ -2662,6 +2674,12 @@ defmodule Enum do
     end
   end
 
+  # TODO: Remove me on v2.0
+  def slice(enumerable, %{__struct__: Range, first: first, last: last} = index_range) do
```

Comment:
I believe we don't need this clause because it will fallback to the protocol.

Reply:
```
iex(7)> Enum.slice([1, 2, 3, 4, 5, 6, 7, 8, 9, 0, 11, 12, 13, 14], old_range)
** (FunctionClauseError) no function clause matching in Enum.slice/2    
    
    The following arguments were given to Enum.slice/2:
    
        # 1
        [1, 2, 3, 4, 5, 6, 7, 8, 9, 0, 11, 12, 13, 14]
    
        # 2
        10..1//-1
    
    Attempted function clauses (showing 1 out of 1):
    
        def slice(enumerable, first..last//step = index_range)
    
    (elixir 1.12.2) lib/enum.ex:2649: Enum.slice/2
```

Reply:
You are correct, here the range is the second argument, so we need to address it!